### PR TITLE
Update the microarray clustering example notebook

### LINKED ISSUE #181: Microarray Analysis Updates: clustering heatmap
Besides the up to date steps listed #173 :

> - Have the getting started template material added to it(# 146)
> - Be updated to adhere to the style guide (#148)
> - Add dataset to S3 buckets.
> - Be added to the Snakefile and rendered with snakemake on the Docker container.

It also needs to have its dataset here (which is n = 4) switched to one with a bigger sample size. 
For the heatmap creation steps, we should also mirror the set up that's used over on rnaseq clustering notebook (make an unannotated heatmap first, then make the annotation data.frame, then make an annotated heatmap).

### PR DESCRIPTION
**Pull request by cbethell**
## Purpose:
What issue(s) does your PR address?

This PR closes #181 (the issue on updating the clustering microarray example notebook).

## Strategy
What was your strategy for this new or edited analysis?

I followed the strategy [noted on the issue](https://github.com/AlexsLemonade/refinebio-examples/issues/181#issue-680193873), which was to:

1. Find a new dataset that's larger than n = 4
2. Have the getting started template material added to it
3. Update to adhere to the style guide
4. Add the dataset added to the S3 bucket. (See https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#adding-datasets-to-the-s3-bucket)
5. Add to the Snakefile and render with snakemake on the Docker container.
6. For the heatmap creation steps, we should also mirror the set up that's used over on rnaseq clustering notebook (make an unannotated heatmap first, then make the annotation data.frame, then make an annotated heatmap).

## Concerns/Questions for reviewers:
What things should reviewers look out for?

- Are there any essential steps missing from the workflow?
- The `set.seed()` step was removed from this notebook because I do not believe it is needed in this notebook, but please correct me if I am wrong.
- Are the instructions/context in the notebook clear and suffice?

### Content checks
* [x] All `{{BLANKS}}` have been replaced with the correct content.
* [x] [Sources are cited](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#citing-sources-in-text)
* [x] Seed is set (if applicable - do not believe it is applicable in this nb)

### Formatting Checks
* [x] Removed any manual numbering of sections.
* [x] Removed any instances of chunk naming.
* [x] Spell checked any Rmd file or md file.
* [x] Comments and documentation are up to date.

### Add datasets to S3
* [x] Added [data and metadata files to S3](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#adding-datasets-to-the-S3-bucket).

### Docker/Snakemake rendering components
* [x] Added the `.html` [link to the navigation bar](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#add-new-analyses-to-the-navbar).
* [x] Any not yet added packages needed for this analysis have been added to the Dockerfile and it successfully builds.
* [x] In the Docker container, [snakemake was run for rendering](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#how-to-re-render-the-notebooks).

### CODE REVIEW

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 1]
```diff
@@ -1,145 +1,347 @@
...
+You can easily switch this to save to a jpeg or tiff by changing the function and file name within the function to the respective file suffix.
+
+```{r}
 # Open a png file
-png(file.path(plots_dir, 
-              "GSE12955_heatmap.png" # Replace file name with a relevant output plot name
-              ))
+png(file.path(
+  plots_dir,
+  "heatmap_GSE24862_non_annotated.png" # Replace file name with a relevant output plot name
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:258]:
```suggestion
  "GSE24862_heatmap_non_annotated.png" # Replace file name with a relevant output plot name
```
Did I miss this elsewhere?

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 2]
```diff
@@ -1,145 +1,347 @@
...
+
+### Save annotated heatmap as a PNG
+
+You can easily switch this to save to a JPEG or TIFF by changing the function and file name within the function to the respective file suffix.
+
+```{r}
+# Open a PNG file
+png(file.path(
+  plots_dir,
+  "heatmap_GSE24862_annotated.png" # Replace file name with a relevant output plot name
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:324]:
```suggestion
  "GSE24862_heatmap_annotated.png" # Replace file name with a relevant output plot name
```

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 3]
```diff
@@ -1,145 +1,347 @@
...
-        bottom_annotation = annot) # assign the previously made annotation object
-
-# Print out the heatmap
-heatmap
+
+To further customize the heatmap, see a vignette for a guide at this [link](https://slowkow.com/notes/pheatmap-tutorial/) [@Slowikowski2017].
+
+```{r}
+# Create and store the heatmap object
+pheatmap <-
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:229]:
```suggestion
heatmap <-
```

We should shy away from using the package name as an object name.

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 4]
```diff
@@ -1,145 +1,347 @@
...
-png(file.path(plots_dir, 
-              "GSE12955_heatmap.png" # Replace file name with a relevant output plot name
-              ))
+png(file.path(
+  plots_dir,
+  "heatmap_GSE24862_non_annotated.png" # Replace file name with a relevant output plot name
+))
+
+# Print your heatmap
+pheatmap
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:262]:
```suggestion
heatmap
```

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 5]
```diff
@@ -1,145 +1,347 @@
...
+  tibble::column_to_rownames("refinebio_accession_code")
+```
+
+### Create annotated heatmap
+
+You can create an annotated heatmap by providing our annotation object to the `annotation_col` argument of the `pheatmap()` function.
+
+```{r}
+# Create and store the annotated heatmap object
+pheatmap_annotated <-
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:294]:
```suggestion
heatmap_annotated <-
```

Comment (cansavvy) [reply 02-microarray/clustering_microarray_01_heatmap.Rmd:294]: Just to make it parallel.

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 6]
```diff
@@ -1,145 +1,347 @@
...
+```{r}
+# Open a PNG file
+png(file.path(
+  plots_dir,
+  "heatmap_GSE24862_annotated.png" # Replace file name with a relevant output plot name
+))
 
 # Print your heatmap
-heatmap
+pheatmap_annotated
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:328]:
```suggestion
heatmap_annotated
```

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 7]
```diff
@@ -1,145 +1,347 @@
...
+<img src="https://github.com/AlexsLemonade/refinebio-examples/raw/e140face75daa6d2c34e30a4755c362e6039a677/template/screenshots/download-email.png" width=500>
+
+It may take a few minutes for the dataset to process.
+You will get an email when it is ready. 
+
+## About the dataset we are using for this example
+
+For this example analysis, we will use this [melanoma cell line dataset](https://www.refine.bio/experiments/GSE24862).
+
+The data that we downloaded from refine.bio for this analysis has 21 microarray samples obtained from three PLX4032-sensitive parental and three PLX4032-resistant sub-lines of melanoma cell lines that were treated or not treated with the RAF-selective inhibitor, PLX4032.
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:83]: I think this sentence has a lot of info in it so its a lot to absorb at once. Can we spread it out so it's a bit easier to follow?

**Review by cansavvy — COMMENTED** [review]
This looks great! Only minor comments.

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 8]
```diff
@@ -1,145 +1,347 @@
...
-creating a heatmap. [@ComplexHeatmap]
+This notebook illustrates one way that you can use microarray data from refine.bio in downstream analyses, specifically in plotting clustered and annotated heatmaps.
 
-```{r Install and attach the ComplexHeatmap library}
-if (!("ComplexHeatmap" %in% installed.packages())) {
-  # Install ComplexHeatmap
-  BiocManager::install("ComplexHeatmap", update = FALSE)
-}
-```
+# How to run this example
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:16]:
```suggestion
⬇️ [**Jump to the analysis code**](#analysis) ⬇️

# How to run this example
```
Very recent add to the remplate #207

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 9]
```diff
@@ -1,145 +1,347 @@
...
+If the concept of a "file path" is unfamiliar to you; we recommend taking a look at our [section about file paths](https://alexslemonade.github.io/refinebio-examples/01-getting-started/getting-started.html#an-important-note-about-file-paths-and-Rmds). 
+
+# Using a different refine.bio dataset with this analysis?
+
+If you'd like to adapt an example analysis to use a different dataset from [refine.bio](https://www.refine.bio/), we recommend placing the files in the `data/` directory you created and changing the filenames and paths in the notebook to match these files (we've put comments to signify where you would need to change the code).
+We suggest saving plots and results to `plots/` and `results/` directories, respectively, as these are automatically created by the notebook.
+From here you can customize this analysis example to fit your own scientific questions and preferences. 
+
+***
+
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:141]:
```suggestion

<!-- Do not delete this line --> <a name="analysis" style="padding-top:56px;margin-top:-56px;">&nbsp;</a>

```

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 10]
```diff
@@ -1,145 +1,347 @@
...
+# Close the png file:
+dev.off()
+```
+
+Now, let's add some annotation bars to our heatmap.
+
+## Prepare metadata for annotation
+
+From the accompanying [paper](https://pubmed.ncbi.nlm.nih.gov/21107323/) [@Nazarian2010], we know that three PLX4032-sensitive parental (P) cell lines (M229, M238 and M249) and three derived PLX4032-resistant (R) sub-lines (M229_r5, M238_r1, and M249_r4) were treated or not treated with the RAF-selective inhibitor, PLX4032.
```
[the remark this answers is not in the document]
Comment (cbethell) [reply 02-microarray/clustering_microarray_01_heatmap.Rmd:272]: It appears that the choice here, in part, had to with these lines not having secondary mutations in the drug target B-RAF (V600E) but I am not sure how they further narrowed down to (1, 4, 5) as they state that this "lack of secondary B-RAF (V600E) mutation along with retention of the original B-RAF(V600E) mutation was confirmed in 16/16 melanoma tumor biopsies".

File: 02-microarray/clustering_microarray_01_heatmap.Rmd (hunk 1 above)
Comment (cbethell) [reply 02-microarray/clustering_microarray_01_heatmap.Rmd]: This was just the format in the [`03-rnaseq/clustering_rnaseq_01_heatmap.Rmd`](https://github.com/AlexsLemonade/refinebio-examples/blob/e8500af03f3ab0fe25863f022d5fe44bd96b27c3/03-rnaseq/clustering_rnaseq_01_heatmap.Rmd#L323) notebook which I adapted some material from to get started on the notebook in this PR. 

I'll propagate these changes to that notebook in this PR as well.

Comment (cansavvy) [reply 02-microarray/clustering_microarray_01_heatmap.Rmd]: Gotcha. I might have missed that there.

File: 02-microarray/clustering_microarray_01_heatmap.Rmd (hunk 7 above)
Comment (cbethell) [reply 02-microarray/clustering_microarray_01_heatmap.Rmd]: I broke up this sentence in the last commit, as suggested. I believe that the language here was the easiest way to word what's going on in the dataset while keeping the sentence relatively short. However, if the wording here still seems like a lot, I can attempt to re-word it, which would likely include adding a sentence or two.

File: 02-microarray/clustering_microarray_01_heatmap.Rmd (hunk 10 above)
[the remark this answers is not in the document]
Comment (cansavvy) [reply 02-microarray/clustering_microarray_01_heatmap.Rmd]: Hmm...  I looked at the paper and their supplementary material and I find their R1, R4, R5 designations confusing. Let's try not to dwell on it too much beyond the idea that R = resistant and that they are different sublines.

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 11]
```diff
@@ -137,9 +270,83 @@ heatmap
...
+    cluster_rows = TRUE,
+    cluster_cols = TRUE,
+    show_rownames = FALSE,
+    annotation_col = annotation_df,
+    main = "Annotated Heatmap",
+    colorRampPalette(c(
+      "deepskyblue",
+      "black",
+      "yellow"
+    ))(25
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:311]: I assume this is how styler wanted it.... odd.

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 12]
```diff
@@ -137,9 +270,83 @@ heatmap
...
+    ),
+    scale = "row" # Scale values in the direction of genes (rows)
+  )
+```
+
+Now that we have annotation bars on our heatmap, we have a better idea of the cell line and treatment groups that appear to cluster together.
+
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:318]: I think we need one sentence here that mentions the results of this heatmap. Mainly that the samples seem to cluster by their cell lines of origin but not so much by whether or not they received the `PLX4302` treatment.

**Review by cansavvy — COMMENTED** [review]
I only have a couple minor comments, and one thing that I think we should contemplate adding but I hadn't thought of before since I hadn't looked at the paper much until now. I think the resistance is a key part of the main question of this experimental model, so it would it might be good to have a `resistant` or `parental` line designation. 

If there isn't a column that already has this info separated out, an added code chunk that uses a `mutate` with a `case_when` that looks for an `_r` should do the trick. Should probably add something about that sometimes there's pertinent information in an experimental model, but that is not always reflected in the refinebio metadata, which is based on author submitted data. But that shouldn't hold back our users from adding a relevant column if they see fit.

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 13]
```diff
@@ -1,134 +1,266 @@
...
+<!-- Do not delete this line --> <a name="analysis" style="padding-top:56px;margin-top:-56px;">&nbsp;</a>
+
+# Clustering Heatmap - Microarray
+
+## Install libraries
+
+See our Getting Started page with [instructions for package installation](https://alexslemonade.github.io/refinebio-examples/01-getting-started/getting-started.html#what-you-need-to-install) for a list of the other software you will need, as well as more tips and resources.
+
+In this analysis, we will be using the R package `pheatmap` for clustering and creating a heatmap  [@Slowikowski2017].
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:153]:
```suggestion
In this analysis, we will be using the R package `pheatmap` for clustering and creating a heatmap [@Slowikowski2017].
```

File: 02-microarray/clustering_microarray_01_heatmap.Rmd [hunk 14]
```diff
@@ -137,9 +269,91 @@ heatmap
...
 
-Print session info:
+Now, let's add some annotation bars to our heatmap.
+
+## Prepare metadata for annotation
 
-```{r Print session info}
-# Print session info 
+From the accompanying [paper](https://pubmed.ncbi.nlm.nih.gov/21107323/), we know that three PLX4032-sensitive parental cell lines (M229, M238 and M249) and three derived PLX4032-resistant (r) sub-lines (M229_r5, M238_r1, and M249_r4) were treated or not treated with the RAF-selective inhibitor, PLX4032 [@Nazarian2010].
+We are going to annotate our heatmap with the variables that hold the `refinebio_cell_line` and `refinebio_treatment` data.
```
Comment (cansavvy) [inline 02-microarray/clustering_microarray_01_heatmap.Rmd:277]: I think this paragraph should be updated to mention that we are going to create a new column from our existing metadata.

**Review by cansavvy — COMMENTED** [review]
One quick thing from before that I think needs to be addressed before we can merge.

I think the paragraph before the metadata set up should be slightly updated to reflect that we are going create the `cell_line_type` variable and we should mention that while this step is very metadata-specific, the users might also find that they need to tailor the metadata for their own needs.

**Review by cansavvy — APPROVED** [review]
Great! LGTM! :shipit:

**Pull request merged**